src/NServiceBus.Persistence.DynamoDB/Saga/SagaPersister.cs

```diff
@@ -255,11 +261,17 @@ public Task Update(IContainSagaData sagaData, ISynchronizedStorageSession sessio
             var map = doc.ToAttributeMap();
             map.Add(configuration.Table.PartitionKeyName, new AttributeValue { S = $"SAGA#{sagaData.Id}" });
             map.Add(configuration.Table.SortKeyName, new AttributeValue { S = $"SAGA#{sagaData.Id}" });  //Sort key
-            // Version should probably be properly moved into metadata to not clash with existing things
-            map.Add(SagaDataVersionAttributeName, new AttributeValue { N = version.ToString() });
+            map.Add(SagaMetadataAttributeName, new AttributeValue
+            {
+                M = new Dictionary<string, AttributeValue>()
+                {
+                    { SagaDataVersionAttributeName, new AttributeValue { N = version.ToString() } },
+                    { "TYPE", new AttributeValue { S = sagaData.GetType().FullName } },
+                    { "SCHEMA_VERSION", new AttributeValue { N = "1"} }
```

Comment:
we could also use a string to contain a MinVer number instead but I assumed that a single number will be enough for any internal handling of different schema versions.

Reply:
I would prefer here at least 1.0 or 1.0.0

Reply:
Can be addressed later